corehq/apps/hqcase/tests/test_explode_cases.py

```diff
@@ -212,3 +214,84 @@ def test_parent_child(self):
         for child in child_cases:
             self.assertEqual(1, len(child.indices))
             self.assertTrue(child.indices[0].referenced_id in parent_cases)
+
+
+class ExplodeExtensionsDBTest(BaseSyncTest):
+
+    def setUp(self):
+        super(ExplodeExtensionsDBTest, self).setUp()
+        self.accessor = CaseAccessors(self.project.name)
+        self._create_case_structure()
+
+    def _create_case_structure(self):
+        """
+                  +----+
+                  | E1 |
+                  +--^-+
+                     |e
+        +---+     +--+-+
+        |O  +--c->| C  |
+        +---+     +--^-+
+                     |e
+                  +--+-+
+                  | E2 |
+                  +----+
+        """
+        case_type = 'case'
+
+        E1 = CaseStructure(
+            case_id='extension_1',
+            attrs={'create': True, 'owner_id': '-'},
+        )
+
+        C = CaseStructure(
+            case_id='child',
+            attrs={'create': True, 'owner_id': '-'},
+            indices=[CaseIndex(
+                E1,
+                identifier='extension_1',
+                relationship='extension',
+                related_type=case_type,
+            )]
+        )
+
+        O = CaseStructure(
+            case_id='owned',
+            attrs={'create': True},
+            indices=[CaseIndex(
+                C,
+                identifier='child',
+                relationship='child',
+                related_type=case_type,
+            )]
+        )
+
+        E2 = CaseStructure(
+            case_id='extension_2',
+            attrs={'create': True, 'owner_id': '-'},
+            indices=[CaseIndex(
+                C,
+                identifier='extension',
+                relationship='extension',
+                related_type=case_type,
+            )]
+        )
+        self.device.post_changes([O, E2])
+
+    def test_case_graph(self):
+        cases = self.device.restore().cases
+        self.accessor.get_case_ids_in_domain
+        graph = CaseGraph(cases)
+        self.assertEqual(
+            ['extension_1', 'child', 'extension_2', 'owned'],
```

Comment:
Correct order for the graph in the diagram is `['owned', 'child', 'extension_2', 'extension_1']` (last 2 are interchangeable)

Reply:
I think I just named these cases in a very confusing manner (copied from a different test in the extension cases sync code). I've updated the names to make things clearer.